File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s081988820_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func solve() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	H, _ := strconv.Atoi(sc.Text())
	sc.Scan()
	W, _ := strconv.Atoi(sc.Text())
	sc.Scan()

	c := make([][]int, 10)
	for i := range c {
		c[i] = make([]int, 10)
	}

	for i := 0; i < 10; i++ {
		for j := 0; j < 10; j++ {
			sc.Scan()
			c[i][j], _ = strconv.Atoi(sc.Text())
		}
	}

	min := make([]int, 10)
	for i := 0; i < 10; i++ {
		min[i] = c[i][1]
	}

	for tc := 0; tc < 10; tc++ {
		for i := 0; i < 10; i++ {
			for j := 0; j < 10; j++ {
				min[i] = minInt(min[i], c[i][j]+min[j])
			}
		}
	}

	ans := 0
	for h := 1; h <= H; h++ {
		for w := 1; w <= W; w++ {
			sc.Scan()
			A, _ := strconv.Atoi(sc.Text())
			if A >= 0 {
				ans += min[A]
			}
		}
	}

	fmt.Println(ans)
}

func minInt(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	solve()
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C++/s081988820_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <optional>

class Main {
public:
    void solve() {
        std::cin >> H >> W;
        c.resize(10, std::vector<int>(10));
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                std::cin >> c[i][j];
            }
        }
        min.resize(10);
        for (int i = 0; i < 10; i++) {
            min[i] = c[i][1];
        }
        for (int tc = 0; tc < 10; tc++) {
            for (int i = 0; i < 10; i++) {
                for (int j = 0; j < 10; j++) {
                    min[i] = std::min(min[i], c[i][j] + min[j]);
                }
            }
        }
        ans = 0;
        for (int h = 1; h <= H; h++) {
            for (int w = 1; w <= W; w++) {
                int A;
                std::cin >> A;
                if (A >= 0) {
                    ans += min[A];
                }
            }
        }
        std::cout << ans << std::endl;
    }

private:
    int H, W, ans;
    std::vector<std::vector<int>> c;
    std::vector<int> min;
};

int main() {
    Main main;
    main.solve();
    return 0;
}
// End of Code
